extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -548,6 +551,32 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
             function _isDisabledRequest(xhr?: XMLHttpRequestInstrumented, request?: Request | string, init?: RequestInit) {
                 let isDisabled = false;
                 let theUrl:string = ((!isString(request) ? ((request ||{}) as Request).url || "" : request as string) ||"").toLowerCase();
+
+                // check excludeRequestFromAutoTrackingPatterns before stripping off any query string
+                arrForEach(_excludeRequestFromAutoTrackingPatterns, (regex: string | RegExp) => {
+                    let isRegExp = false;
+                    if (!isString(regex)) {
+                        isRegExp = true;
+                        const regexObj = {flags: (regex as any).flags, source: (regex as any).source};
+                        regex = JSON.stringify(regexObj);
+                    }
+                    if (!isDisabled) {
+                        if (isRegExp) {
+                            const revertRegexObj = JSON.parse(regex as string);
+                            const regexp = new RegExp(revertRegexObj.source, revertRegexObj.flags);
+                            isDisabled = regexp.test(theUrl);
+                        } else {
+                            const regexp = new RegExp(regex);
+                            isDisabled = regexp.test(theUrl);
+                        }
+                    }
```

Comment:
Got the convert RegExp to string then back method from this [answer](https://stackoverflow.com/questions/39154255/converting-regexp-to-string-then-back-to-regexp).
Doc points the RegExp does have [toString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/toString) method but it converts a regex to a string directly and if convert it back it becomes `"\\xxx\\"`. 
@MSNev do you think it's better to use `JSON.stringify` or to trim the extra `"\"` with `toString` method or there's a better way? Thanks~